src/plugins/visualize/public/application/components/visualize_listing.tsx

```diff
@@ -0,0 +1,159 @@
+    },
+  } = useKibana<VisualizeServices>();
+  const { pathname } = useLocation();
+  const closeNewVisModal = useRef(() => {});
+  const listingLimit = savedObjectsPublic.settings.getListingLimit();
+
+  useEffect(() => {
+    if (pathname === '/new') {
+      // In case the user navigated to the page via the /visualize/new URL we start the dialog immediately
+      closeNewVisModal.current = visualizations.showNewVisModal({
+        onClose: () => {
+          // In case the user came via a URL to this page, change the URL to the regular landing page URL after closing the modal
+          history.push(VisualizeConstants.LANDING_PAGE_PATH);
+        },
+      });
+    } else {
+      // close modal window if exists
+      closeNewVisModal.current();
+    }
+  }, [history, pathname, visualizations]);
+
+  useMount(() => {
+    chrome.setBreadcrumbs([
+      {
+        text: i18n.translate('visualize.visualizeListingBreadcrumbsTitle', {
+          defaultMessage: 'Visualize',
+        }),
+      },
+    ]);
+    chrome.docTitle.change(
+      i18n.translate('visualize.listingPageTitle', { defaultMessage: 'Visualize' })
+    );
+  });
+  useUnmount(() => closeNewVisModal.current());
+
+  const createNewVis = useCallback(() => {
+    closeNewVisModal.current = visualizations.showNewVisModal();
+  }, [visualizations]);
+
+  const editItem = useCallback(
+    ({ editUrl, editApp }) => {
+      if (editApp) {
+        application.navigateToApp(editApp, { path: editUrl });
+        return;
+      }
+      // for visualizations the edit and view URLs are the same
+      history.push(editUrl);
+    },
+    [application, history]
+  );
+
+  const noItemsFragment = useMemo(() => getNoItemsMessage(createNewVis), [createNewVis]);
+  const tableColumns = useMemo(() => getTableColumns(application, history), [application, history]);
+
+  const fetchItems = useCallback(
+    (filter) => {
+      const isLabsEnabled = uiSettings.get(VISUALIZE_ENABLE_LABS_SETTING);
+      return savedVisualizations
+        .findListItems(filter, listingLimit)
+        .then(({ total, hits }: { total: number; hits: object[] }) => ({
+          total,
+          hits: hits.filter((result: any) => isLabsEnabled || result.type.stage !== 'experimental'),
+        }));
+    },
+    [listingLimit, savedVisualizations, uiSettings]
+  );
+
+  const deleteItems = useCallback(
+    async (selectedItems: object[]) => {
+      await Promise.all(
+        selectedItems.map((item: any) => savedObjects.client.delete(item.savedObjectType, item.id))
+      ).catch((error) => {
+        toastNotifications.addError(error, {
+          title: i18n.translate('visualize.visualizeListingDeleteErrorTitle', {
+            defaultMessage: 'Error deleting visualization',
+          }),
+        });
+      });
+    },
+    [savedObjects.client, toastNotifications]
+  );
+
+  return (
+    <div
+      className="kuiViewContent kuiViewContent--constrainedWidth kuiViewContentItem"
```

Comment:
These classes are part of the legacy UI framework - can we move away from them? It also seems like these are the last usages of `kui*` stuff within visualize.

Reply:
Thanks for highlighting this! I intentionally left it here to ask @cchaos for suggestion here! Maybe necessary classes already exist and I can just use them, or create new (common for other apps or internal for visualize only)

Reply:
Yep, all these classes are completely unnecessary. And unless the you **really** need the DTS to be on a wrapper around the `TableListView` and  not on it, then you can remove the div completley.

Reply:
Thanks @cchaos ,
removed the wrapper since the DTS is duplicated inside the `TableListView` component!